OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnProduction.cs

```diff
@@ -0,0 +1,73 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2018 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Mods.Common.Activities;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Grants a condition when this actor produces a specific actor.")]
+	public class GrantConditionOnProductionInfo : ITraitInfo
+	{
+		[GrantedConditionReference]
+		[Desc("The condition to grant")]
+		public readonly string Condition = null;
+
+		[ActorReference]
+		[Desc("The actors to grant condition for. If empty condition will be granted for all actors.")]
+		public readonly HashSet<string> Actors = new HashSet<string>();
+
+		public object Create(ActorInitializer init) { return new GrantConditionOnProduction(init.Self, this); }
+	}
+
+	public class GrantConditionOnProduction : INotifyCreated, INotifyOtherProduction
```

Comment:
Why not `INotifyProduction`?

Reply:
I didn't realise we already had such interface, yeah probably a better idea and actually is what i was looking for.